query/src/lib.rs

```diff
@@ -21,13 +28,15 @@ pub struct ParsedQuery {
 
 /// Handles the registration and invocation of UDF queries in DataFusion with a
 /// pre-compiled WASM component.
-pub struct UdfQueryParser<'a> {
+pub struct UdfQueryParser<'a, F: UdfCodeFormatter> {
     /// Pre-compiled WASM component.
     /// Necessary to create UDFs.
     components: HashMap<String, &'a WasmComponentPrecompiled>,
+    /// Code formatter for UDF code
```

Comment:
Your formatter is language-specific, so you should have one per language. I suggest you don't add this a generic (in general generics are to be avoided due to the code bloat they produce anyways) and design it a bit like this:

```rust
languages: HashMap<String, Lang<'a>>
```

and then

```rust
struct Lang<'a> {
    component: &'a WasmComponentPrecompiled,
    formatter: Box<dyn UdfCodeFormatter>,
}
```